src/PollingBlockTracker.ts

```diff
@@ -89,7 +91,19 @@ export class PollingBlockTracker
   async destroy() {
     this._cancelBlockResetTimeout();
     this._maybeEnd();
-    super.removeAllListeners();
+    this.eventNames().forEach((eventName) =>
```

Comment:
Do we want to explain why excluding internal listeners are necessary? Perhaps something like:

``` suggestion
    // The `getLatestBlock` method waits until the next iteration of the polling
    // loop fetches the latest block. It waits for this block to arrive by
    // listening to the `latest` event, and it handles errors by listening to
    // the `error` event. Because `getLatestBlock` is actively relying on these
    // listeners to get called in order to resolve or reject a promise, we don't
    // want to remove them (otherwise the polling loop will run forever and the
    // promise will never get fulfilled). We will handle removing them manually.
    this.eventNames().forEach((eventName) =>
```

Reply:
By the way, I was playing around with this PR and it seems that if you revert the changes to `destroy` and `_getBlockTrackerEventCount`, then your test still works. It seems that listening to `error` solved at least one bug. But perhaps there is another test we need to write to verify the bug we are trying to solve is fixed? I can take another look tomorrow to see what that would be (or, it may be what you suggested elsewhere, I'm not sure).

Reply:
Your comment was referring to a change that has been reverted. Moreover, I noticed that I unintentionally used `this.removeAllListeners`, while `destroy` originally used `super` instead. 

Since `super.removeAllListeners()` also removes internal listeners, and `destroy()` also rejects an eventual pending promise, do you think we can resolve this conversation or are there other things to consider?

Reply:
I feel like it's worth documenting the decisions we made here somewhere and why we have to keep track of and exclude internal listeners, because we will probably look back on this at some point in the future and want to understand this again. But I can't think of a good way to do this right now since there are multiple places where we refer to internal listeners in the code. The PR description also does a good job of explaining the problem, so maybe that is sufficient for now.

I will resolve this conversation.